src/components/PeopleTable.tsx

```diff
@@ -1,4 +1,78 @@
-export const PeopleTable = () => {
+import { FC } from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+import { PersonLink } from './PersonLink';
+import { SortBy } from '../types/SortBy';
+import { SortLink } from './SortLink';
+
+type Props = {
+  people: Person[],
+};
+
+export const PeopleTable: FC<Props> = ({ people }) => {
+  const { slug } = useParams();
+  const [searchParams] = useSearchParams();
+
+  const sex = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries');
+  const query = searchParams.get('query')?.toLowerCase() || '';
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  const filteredPeople = people.filter(person => {
+    if (sex && person.sex !== sex) {
+      return false;
+    }
+
+    if (centuries.length) {
+      const century = Math.ceil(person.born / 100);
+
+      if (!centuries.includes(`${century}`)) {
+        return false;
+      }
+    }
+
+    if (query
+      && !(
+        person.name.toLowerCase().includes(query)
+        || person.motherName?.toLowerCase().includes(query)
+        || person.fatherName?.toLowerCase().includes(query)
+      )
+    ) {
+      return false;
+    }
+
+    return true;
+  });
+
+  const onSort = () => {
+    switch (sort) {
+      case SortBy.name:
+      case SortBy.sex:
+        filteredPeople.sort((person1, person2) => person1[sort]
+          .localeCompare(person2[sort]));
+        break;
+
+      case SortBy.born:
+      case SortBy.died:
+        filteredPeople
+          .sort((person1, person2) => person1[sort] - person2[sort]);
+        break;
+
+      default:
+        filteredPeople.sort((person1, person2) => person1.slug
+          .localeCompare(person2.slug));
+    }
+
+    if (order) {
+      filteredPeople.reverse();
+    }
+  };
+
+  if (sort) {
```

Comment:
It is better to call the 'onSort()' function inside a useEffect hook to avoid unnecessary re-renders

Reply:
not fixed. we can avoid useEffect and instead wrap onSort in useMemo and return sortedPeople. also better to wrap filteredPeople in useMemo
```
  const filteredPeople = useMemo(() => {
    return people.filter(person => {
      if (sex && person.sex !== sex) {
        return false;
      }

      if (centuries.length) {
        const century = Math.ceil(person.born / 100);

        if (!centuries.includes(`${century}`)) {
          return false;
        }
      }

      if (query
        && !(
          person.name.toLowerCase().includes(query)
          || person.motherName?.toLowerCase().includes(query)
          || person.fatherName?.toLowerCase().includes(query)
        )
      ) {
        return false;
      }

      return true;
    });
  }, [people, query, sort, order]);

  ```
    const sortedPeople = useMemo(() => {
      const sorted = [...filteredPeople];

    switch (sort) {
      case SortBy.name:
      case SortBy.sex:
        sorted.sort((person1, person2) => person1[sort]
          .localeCompare(person2[sort]));
        break;

      case SortBy.born:
      case SortBy.died:
        sorted
          .sort((person1, person2) => person1[sort] - person2[sort]);
        break;

      default:
        sorted.sort((person1, person2) => person1.slug
          .localeCompare(person2.slug));
    }

    if (order) {
      sorted.reverse();
    }

      return sorted;
    }, [order, sort, filteredPeople]);